### PR TITLE
Expose failed pod logs

### PR DESCRIPTION
**Pull request by bkeroackdsc**
When a chart group install or upgrade fails, this attempts to figure out what Kubernetes objects caused the failure and then returns that information (along with pod logs) to the metahelm client in the form of a ChartError struct.

In particular, this is intended to solve for the situation where a chart install times out due to one or more pods going into CrashLoopBackoff, due to application errors or other problems. This PR examines the release manifest to determine what higher level k8s objects were created (Deployments, Jobs, DaemonSets) and then queries k8s to find the pods associated with those objects, finds the failed ones and pulls the logs.

In the event of Helm errors unrelated to application failure, a normal error or empty ChartError is returned.

Notes:
- This only detects and prints the first error detected. For example, if three charts are being installed simultaneously, the chart that returns an error first will be inspected for failed resources. The others may fail as well but will not be inspected.
- When Helm is being used to determine chart health (equivalent to `helm install --wait`), Helm does not wait for non-hook jobs to complete successfully. Therefore a failed job alone will not be detected, however if a deployment *and* a job fail, both would be detected and returned to the client because the deployment failure would cause Helm to return an error.
- Due to https://github.com/helm/helm/issues/4210 , `extension/v1beta1/Deployment` does not work with Helm `--wait`. Use `apps/v1/Deployment` instead.

### CODE REVIEW

File: pkg/metahelm/error.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+				fp.Phase = string(pod.Status.Phase)
+				fp.Message = pod.Status.Message
+				fp.Reason = pod.Status.Reason
+				fp.Conditions = pod.Status.Conditions
+				fp.ContainerStatuses = pod.Status.ContainerStatuses
+				// get logs
+				for _, cs := range fp.ContainerStatuses {
+					if !cs.Ready && cs.LastTerminationState.Terminated != nil && cs.LastTerminationState.Terminated.ExitCode != 0 {
+						plopts.Container = cs.Name
+						req := kc.CoreV1().Pods(rls.Namespace).GetLogs(pod.Name, &plopts)
```
Comment (jmedberydsc) [inline pkg/metahelm/error.go]: 👍 I thought there was more to this, but it was just a simple as passing in the max number of lines

**Comment by bkeroackdsc** [comment]
@jmedberydsc I appreciate the 👍 but this is still very much a WIP and not ready for CR yet. :)

File: pkg/metahelm/error.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	deploymentutil "k8s.io/kubernetes/pkg/controller/deployment/util"
+)
+
+// FailedPod models a single failed pod with metadata and logs
+type FailedPod struct {
+	Name, Phase, Message, Reason string
+	Conditions                   []corev1.PodCondition
+	ContainerStatuses            []corev1.ContainerStatus
+	// Logs is a map of container name to raw log (stdout/stderr) output
+	Logs map[string][]byte
```
Comment (esterlinej) [inline pkg/metahelm/error.go]: struct param type alignment

File: pkg/metahelm/error.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+// ChartError is a chart install/upgrade error due to failing Kubernetes resources. It contains all Deployment, Job or DaemonSet-related pods that appear to
+// be in a failed state, including up to MaxPodLogLines of log data for each.
+type ChartError struct {
```
Comment (esterlinej) [inline pkg/metahelm/error.go]: struct param alignment, not to block, just noting

**Pull request merged**